### ISSUE TITLE
Using `destroySession` does not always remove the session

### ISSUE THREAD

**Issue by charklewis** [remix-run/remix#225]
I have noticed when using multiple sessions that the `destorySession` method from `createCookieSessionStorage` doesn't destroy the session. It does correctly set the header, however the browser keeps the prior value. 

I am updating the headers in the below manner.

```
const headers: any = new Headers()
headers.append("Set-Cookie", await destorySession(session))
headers.append("Set-Cookie", await commitAnotherSession(anotherSession))
```

**Comment by machour** [comment]
@charklewis We need a reproduction case for this issue, as it's too vague and the problem could be caused by multiple factors (like two loaders called in parallel, or whatever could cause a race condition).

If this is still a problem for you, please open a new issue providing a link to a repository showing the problem 🙏🏼
